zerver/tests/test_realm_export.py

```diff
@@ -0,0 +1,104 @@
+from mock import patch
+
+from django.test import override_settings
+from django.conf import settings
+from django.utils.timezone import now as timezone_now
+
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.test_helpers import use_s3_backend, create_s3_buckets
+from zerver.views.public_export import public_only_realm_export
+from zerver.models import RealmAuditLog, UserProfile
+
+# TODO: Mock export_realm_wrapper to test for s3 or local
+class RealmExportTest(ZulipTestCase):
+    def setUp(self) -> UserProfile:
+        admin = self.example_user('iago')
+        self.login(admin.email)
+        return admin
+
+    def test_export_as_not_admin(self) -> None:
+        user = self.example_user('hamlet')
+        self.login(user.email)
+        with self.assertRaises(JsonableError):
+            public_only_realm_export(self.client_post, user)
+
+    @use_s3_backend
+    def test_endpoint_s3(self) -> None:
+        admin = self.setUp()
```

Comment:
This function is called automatically even if you don't call it; read about `setUp` methods in the Python unit test system.  I redid this to not have this weird structure, though further cleanup would be useful.